stock_picking_transfer_lot_autoassign/models/stock_move.py

```diff
@@ -0,0 +1,23 @@
+# Copyright 2017 Pedro M. Baeza <pedro.baeza@tecnativa.com>
+# Copyright 2018 David Vidal <david.vidal@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models
+
+
+class StockMove(models.Model):
+    _inherit = 'stock.move'
+
+    def _prepare_move_line_vals(self, quantity=None, reserved_quant=None):
+        """Auto-assign as done the quantity proposed for the lots"""
+        self.ensure_one()
+        res = super(StockMove, self)._prepare_move_line_vals(
+            quantity, reserved_quant,
+        )
+        if (self.picking_id.picking_type_id.avoid_internal_assignment or
+                not res.get('lot_id')):
```

Comment:
this is a strange and asymetric behavior. If I have both tracked and untracked products in a picking, then the update of qty_done will only happen for tracked product, which will result in a backorder if I just click on "Validate" without changing the done quantities, which seems to break the principle of least surprise.

Reply:
Perhaps we can add an option in picking type to allow choose if autoassign all (tracked and un tracked products), or directly autoassign all...
@chienandalu @pedrobaeza What is your opinion about this?